00-Login-Kt/app/src/main/java/com/auth0/sample/MainActivity.kt

```diff
@@ -96,11 +136,71 @@ class MainActivity : AppCompatActivity() {
                     }
 
                     override fun onSuccess(profile: UserProfile?) {
-                        binding.userProfile.setText(
-                                "Name: ${profile?.name}\n" +
-                                "Email: ${profile?.email}")
+                        cachedUserProfile = profile;
+                        updateUI()
                     }
         }) }
     }
 
+    private fun getUserMetadata() {
+        // Get the access token so that we can make calls to the management API
+        cachedCredentials?.accessToken?.let { accessToken ->
+            // Get the user ID and call the full getUser Management API endpoint, to retrieve the full profile information
+            cachedUserProfile?.getId()?.let { userId ->
+                // Create the user API client
+                val usersClient = UsersAPIClient(account, accessToken)
+
+                // Get the full user profile
+                usersClient.getProfile(userId).start(object: Callback<UserProfile, ManagementException> {
+                    override fun onFailure(exception: ManagementException) {
+                        Snackbar.make(
```

Comment:
what are your thoughts on extracting this snackbar creation (not just for error) to a separate function that receives only the string/message? either here or in the quickstart article

Reply:
Makes sense, let me put that in.